extensions/amp-intersection-observer-polyfill/0.1/cross-origin-observer.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MessageType} from '../../../src/3p-frame-messaging';
+import {WindowInterface} from '../../../src/window-interface';
+import {getMode} from '../../../src/mode';
+import {iframeMessagingClientFor} from '../../../src/inabox/inabox-iframe-messaging-client';
+import {layoutRectLtwh} from '../../../src/layout-rect';
+
+/**
+ * Setup cross-origin iframe polyfill for AMP iframes, such as inabox.
+ *
+ * @param {!Window} win
+ */
+export function maybeSetupCrossOriginObserver(win) {
+  if (win == WindowInterface.getTop(win) || getMode(win).runtime != 'inabox') {
+    // Not an iframe at all.
+    return;
+  }
+
+  // Check if there is an iframe client connected and if it's indeed a polyfill.
+  const iframeClient = iframeMessagingClientFor(win);
+  const setupPolyfillUpdater =
+    win.IntersectionObserver['_setupCrossOriginUpdater'];
```

Comment:
Where is `_setupCrossOriginUpdater` at?

Reply:
It's defined in the polyfill [here](https://github.com/w3c/IntersectionObserver/blob/60abaeacca6ceb039161863f24c98bcde8aeebfc/polyfill/intersection-observer.js#L172).

Reply:
And I can confirm: the declaration and caller are both compiled correctly.